### ISSUE TITLE
Docker file fails to build: Unbound value Input_channel.read_one_chunk_at_a_time

### ISSUE THREAD

**Issue by Kakadu** [ocaml-multicore/retro-httpaf-bench#25]
Could you double check?

```
➜  retro-httpaf-bench git:(master) make                                                                                                                                       4.12.0+domains
mkdir -p build
cd cohttp-eio; docker build  -t cohttp-eio .
Sending build context to Docker daemon  972.8kB
Step 1/8 : FROM ocaml/opam:debian-11-ocaml-4.12-domains AS cohttp-eio
 ---> 51c953fd927d
Step 2/8 : RUN (cd opam-repository && git pull origin 75f69f2010f203d29adb423858f93bf4406c7869 && opam update)
 ---> Using cache
 ---> 3e091d5e8a1c
Step 3/8 : WORKDIR /src
 ---> Using cache
 ---> 5eefa7bf5011
Step 4/8 : RUN opam depext -i ppx_cstruct dune fmt logs cstruct faraday mtime optint lwt-dllist psq luv luv_unix uri ppx_sexp_conv eio_main base64 re sexplib0 ppx_sexp_conv stringext fmt uri-sexp
 ---> Using cache
 ---> 5b05df3e0f30
Step 5/8 : COPY --chown=opam . /src
 ---> Using cache
 ---> 7e2f1ea122d3
Step 6/8 : RUN sudo chown opam .
 ---> Using cache
 ---> 820b8d002448
Step 7/8 : RUN opam exec -- dune build --profile=release
 ---> Using cache
 ---> 7a54c046373d
Step 8/8 : RUN sudo mv /src/_build/default/cohttp_eio.exe /src/cohttp-eio.exe
 ---> Using cache
 ---> 2abe5df122e5
Successfully built 2abe5df122e5
Successfully tagged cohttp-eio:latest
cd cohttp-eio; docker create -ti --name cohttp-eio-c cohttp-eio:latest
49654d5b78a2df27f5669e9ab3d3d53b093d2d595bacda9f4f2c11ec59ca4f5b
cd cohttp-eio; docker cp cohttp-eio-c:/src/cohttp-eio.exe ../build/cohttp_eio.exe
cd cohttp-eio; docker rm cohttp-eio-c
cohttp-eio-c
cd httpaf-eio; docker build  -t httpaf-eio .
Sending build context to Docker daemon  3.449MB
Step 1/8 : FROM ocaml/opam:debian-11-ocaml-4.12-domains AS eio
 ---> 51c953fd927d
Step 2/8 : RUN (cd opam-repository && git pull origin 75f69f2010f203d29adb423858f93bf4406c7869 && opam update)
 ---> Using cache
 ---> 3e091d5e8a1c
Step 3/8 : WORKDIR /src
 ---> Using cache
 ---> 5eefa7bf5011
Step 4/8 : RUN opam depext -i ppx_cstruct dune fmt logs cstruct faraday mtime optint lwt-dllist psq luv eio_main
 ---> Using cache
 ---> cbb94bd2c2cf
Step 5/8 : COPY --chown=opam . /src
 ---> Using cache
 ---> 90bf2166ec49
Step 6/8 : RUN sudo chown opam .
 ---> Using cache
 ---> d91db83e8d28
Step 7/8 : RUN opam exec -- dune build --profile=release
 ---> Using cache
 ---> b645315d9328
Step 8/8 : RUN sudo mv /src/_build/default/wrk_effects_benchmark.exe /src/httpaf-eio.exe
 ---> Using cache
 ---> 530522d2c15d
Successfully built 530522d2c15d
Successfully tagged httpaf-eio:latest
cd httpaf-eio; docker create -ti --name httpaf-eio-c httpaf-eio:latest
bef70e66b801f977377f99e9303d2fa5f5287362f1e49103845edbeddcb89f92
cd httpaf-eio; docker cp httpaf-eio-c:/src/httpaf-eio.exe ../build/httpaf_eio.exe
cd httpaf-eio; docker rm httpaf-eio-c
httpaf-eio-c
cd httpaf-effects; docker build  -t httpaf-effects .
Sending build context to Docker daemon  13.82kB
Step 1/9 : FROM ocaml/opam:debian-11-ocaml-4.12-domains-effects AS httpaf-effects
 ---> a16ca86ffce7
Step 2/9 : RUN sudo ln -f /usr/bin/opam-2.1 /usr/bin/opam
 ---> Using cache
 ---> 9b3a9a899f2c
Step 3/9 : RUN opam pin -n aeio git+https://github.com/kayceesrk/ocaml-aeio.git
 ---> Using cache
 ---> 868eb58e1915
Step 4/9 : RUN opam install -y conf-libev httpaf lwt dune aeio
 ---> Using cache
 ---> 6a02dd246a84
Step 5/9 : WORKDIR /src
 ---> Using cache
 ---> 20c17871b12c
Step 6/9 : COPY --chown=opam . /src
 ---> Using cache
 ---> 0633a2c99baf
Step 7/9 : RUN sudo chown opam .
 ---> Using cache
 ---> 2a30469c0867
Step 8/9 : RUN opam exec -- dune build --profile=release
 ---> Using cache
 ---> 9614baa2768e
Step 9/9 : RUN sudo mv /src/_build/default/wrk_effects_benchmark.exe /src/httpaf-effects.exe
 ---> Using cache
 ---> 44b81c082c6b
Successfully built 44b81c082c6b
Successfully tagged httpaf-effects:latest
cd httpaf-effects; docker create -ti --name httpaf-effects-c httpaf-effects:latest
25a9c062cd24859128d8e25a27a35c781006c91d0e65832f13d8071f7451c760
cd httpaf-effects; docker cp httpaf-effects-c:/src/httpaf-effects.exe ../build/httpaf_effects.exe
cd httpaf-effects; docker rm httpaf-effects-c
httpaf-effects-c
cd cohttp-lwt-unix; docker build  -t cohttp-lwt-unix .
Sending build context to Docker daemon  8.192kB
Step 1/9 : FROM ocaml/opam:debian-11-ocaml-4.12 AS ocaml-412
 ---> 1611801eac72
Step 2/9 : RUN sudo ln -f /usr/bin/opam-2.1 /usr/bin/opam
 ---> Using cache
 ---> b95df4e2efc1
Step 3/9 : RUN opam install -y conf-libev
 ---> Using cache
 ---> b1c83a0ec354
Step 4/9 : RUN opam pin --with-version dev git+https://github.com/mirage/ocaml-cohttp.git -y
 ---> Using cache
 ---> cf2f58e5a5fa
Step 5/9 : WORKDIR /src
 ---> Using cache
 ---> c6487c1ad9a4
Step 6/9 : COPY --chown=opam . /src
 ---> Using cache
 ---> 667ea1511886
Step 7/9 : RUN sudo chown opam .
 ---> Using cache
 ---> 983c7c8fc70a
Step 8/9 : RUN opam exec -- dune build --profile=release
 ---> Using cache
 ---> 3d9f31abd3e0
Step 9/9 : RUN sudo mv /src/_build/default/cohttp_lwt_unix.exe /src/cohttp-lwt-unix.exe
 ---> Using cache
 ---> 2a38992e50d6
Successfully built 2a38992e50d6
Successfully tagged cohttp-lwt-unix:latest
cd cohttp-lwt-unix; docker create -ti --name cohttp-lwt-unix-c cohttp-lwt-unix:latest
3281ae754520978a0ebb067cdeeb6e8f316a85e9b599e4ec52df0ce0628be7b7
cd cohttp-lwt-unix; docker cp cohttp-lwt-unix-c:/src/cohttp-lwt-unix.exe ../build/cohttp_lwt_unix.exe
cd cohttp-lwt-unix; docker rm cohttp-lwt-unix-c
cohttp-lwt-unix-c
cd httpaf-lwt; docker build  -t httpaf-lwt .
Sending build context to Docker daemon  9.216kB
Step 1/8 : FROM ocaml/opam:debian-11-ocaml-4.12 AS ocaml-412
 ---> 1611801eac72
Step 2/8 : RUN sudo ln -f /usr/bin/opam-2.1 /usr/bin/opam
 ---> Using cache
 ---> b95df4e2efc1
Step 3/8 : RUN opam install -y conf-libev core lwt httpaf httpaf-lwt-unix
 ---> Using cache
 ---> 1e16476092fa
Step 4/8 : WORKDIR /src
 ---> Using cache
 ---> 7252a0a0403b
Step 5/8 : COPY --chown=opam . /src
 ---> Using cache
 ---> 26aff2ea11b1
Step 6/8 : RUN sudo chown opam .
 ---> Using cache
 ---> 217d000a0c5e
Step 7/8 : RUN opam exec -- dune build --profile=release
 ---> Using cache
 ---> 040ffd1e226f
Step 8/8 : RUN sudo mv /src/_build/default/httpaf_lwt.exe /src/httpaf-lwt.exe
 ---> Using cache
 ---> fd0b47fb3717
Successfully built fd0b47fb3717
Successfully tagged httpaf-lwt:latest
cd httpaf-lwt; docker create -ti --name httpaf-lwt-c httpaf-lwt:latest
7b640848016a86ddeebeb99a45178917e5931b9c2812179d0ec4de41f7dce001
cd httpaf-lwt; docker cp httpaf-lwt-c:/src/httpaf-lwt.exe ../build/httpaf_lwt.exe
cd httpaf-lwt; docker rm httpaf-lwt-c
httpaf-lwt-c
cd httpaf-shuttle-async; docker build  -t httpaf-shuttle-async .
Sending build context to Docker daemon  12.29kB
Step 1/8 : FROM ocaml/opam:debian-11-ocaml-4.12 AS ocaml-412
 ---> 1611801eac72
Step 2/8 : RUN sudo ln -f /usr/bin/opam-2.1 /usr/bin/opam
 ---> Using cache
 ---> b95df4e2efc1
Step 3/8 : RUN opam install -y core httpaf shuttle async
 ---> Using cache
 ---> 71b5a7bfc713
Step 4/8 : WORKDIR /src
 ---> Using cache
 ---> 67d5a80071c6
Step 5/8 : COPY --chown=opam . /src
 ---> Using cache
 ---> e396b7a35ba7
Step 6/8 : RUN sudo chown opam .
 ---> Using cache
 ---> 0dd5159d9b86
Step 7/8 : RUN opam exec -- dune build --profile=release
 ---> Running in fd0c80da3d5a
File "protocol.ml", line 13, characters 8-41:
13 |         Output_channel.schedule_bigstring writer buffer ~pos:off ~len;
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Alert deprecated: Shuttle.Output_channel.schedule_bigstring
schedule_bigstring will be removed in a future release. Use [write_bigstring] instead.
File "protocol.ml", line 37, characters 8-46:
37 |         Input_channel.read_one_chunk_at_a_time reader ~on_chunk:(fun buf ->
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: Unbound value Input_channel.read_one_chunk_at_a_time
The command '/bin/sh -c opam exec -- dune build --profile=release' returned a non-zero code: 1
make: *** [Makefile:40: httpaf-shuttle-async] Error 1
```

**Comment by talex5** [comment]
CI now shows some build problems: https://github.com/ocaml-multicore/retro-httpaf-bench/runs/5773640002?check_suite_focus=true, as does #24.

**Comment by anuragsoni** [comment]
```
Alert deprecated: Shuttle.Output_channel.schedule_bigstring
schedule_bigstring will be removed in a future release. Use [write_bigstring] instead.
File "protocol.ml", line 37, characters 8-46:
37 |         Input_channel.read_one_chunk_at_a_time reader ~on_chunk:(fun buf ->
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
This is related to some breaking changes I made within shuttle. #24 is supposed to resolve this as it replaces the `shuttle` based http server in this repository to instead use  the shuttle based `http_async` that was published to the opam-repository a few weeks ago.

**Comment by talex5** [comment]
Fixed by #24 - thanks @anuragsoni!
